SCAutolib/models/file.py

```diff
@@ -0,0 +1,419 @@
+"""
+This module contains classes that represent configuration files. Each class
+contains information and methods to manipulate specific config file except
+the parent (File) class that is supposed to operate on general config file.
+
+basic operations on config files defined in this module:
+    create: create content of config file usually based on template.
+            Note, that some child classes may also update content of config
+            file if it already existed and backup original file.
+    set:    modify values of config files, add keys or sections
+            if necessary
+    save:   save modified content to config file
+    clean:  remove config file; note that some child classes may also restore
+            original config file if backup exists.
+"""
+from configparser import ConfigParser
+from pathlib import Path
+from shutil import copy2
+from typing import Union
+
+from SCAutolib import logger
+from SCAutolib import TEMPLATES_DIR, LIB_BACKUP
+
+
+class File:
+    """
+    This class defines an interface for generic operations on config files
+
+    create: create content of config file based on template file
+    set:    modify content of config files, add keys or sections if necessary
+    save:   save config file
+    clean:  remove config file
+
+    Note:   Set method operates on 1) files compatible with ConfigParser (i.e.
+            files containing sections); 2) simple config files without sections.
+            Other formats of config files are not supported.
+    """
+    _conf_file = None
+    _template = None
+    _default_parser = None
+    _simple_content = None
+
+    def __init__(self, filepath: Union[str, Path], template: str = None):
+        """
+        Init of File
+
+        :param filepath: Path of config file
+        :type filepath: str or pathlib.Path
+        :param template: Path of template file
+        :type template: str
+        """
+        self._conf_file = Path(filepath)
+        if template is not None:
+            self._template = Path(template)
+
+    def create(self):
+        """
+        Populate internal parser object with content based on template.
+        """
+        if self._conf_file.exists():
+            logger.warning(f"Create error: {self._conf_file} already exists.")
+            raise FileExistsError(f'{self._conf_file} already exists')
+        else:
+            self._default_parser = ConfigParser()
+            self._default_parser.optionxform = str
+            if self._template is None:
+                raise FileNotFoundError("Template file was not provided.")
```

Comment:
I don't think a template is a must, no? The user can choose to modify a file without providing a template.